Source/NSubstitute/Core/CallResults.cs

```diff
@@ -6,12 +6,18 @@ namespace NSubstitute.Core
     public class CallResults : ICallResults
     {
         readonly ICallInfoFactory _callInfoFactory;
+        private readonly bool _skipVoidMethods;
         ConcurrentQueue<ResultForCallSpec> _results;
 
-        public CallResults(ICallInfoFactory callInfoFactory)
+        public CallResults(ICallInfoFactory callInfoFactory, bool skipVoidMethodsForPerformance)
```

Comment:
> But you cannot call it for void method, only WhenDo and CallActions work for them.

Yep, I noticed that. For me it looked like just an API limitation, but it seems to be a part of god's plan :) Deeply inside I still don't agree with it and for me there is no difference (`out` is essentially the same `return`, introduced to pass more than one value) , but I understand your point and accept it :hushed:.

I'll mention you and will ask for your review when I implement integration. Let me know if I misunderstood you. Hope, that is not because you noticed that I'm going to make some horrible decisions (though, likely, that is an exact reason given all the discussion above) :disappointed: 😉